unit_tests/test_rest_asset_client.py

```diff
@@ -0,0 +1,297 @@
+
+TEST_COMMAND_RESPONSE = {"command": "register",
+                         "transaction_id": TEST_TRANSACTION_ID,
+                         "registered": True
+                         }
+
+
+@pytest.fixture(name="mock_connections", autouse=True)
+def fixture_mock_connections(mocker):
+    # gsy_framework/client_connections/utils.py
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.retrieve_jwt_key_from_server")
+    mocker.patch("gsy_framework.client_connections.utils.RepeatingTimer")
+
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.WebsocketThread")
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.ThreadPoolExecutor")
+
+
+@pytest.fixture(name="mock_environment_use_functions")
+def fixture_mock_env_use_functions(mocker):
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.simulation_id_from_env",
+                 return_value=TEST_SIMULATION_ID)
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.domain_name_from_env",
+                 return_value=TEST_DOMAIN_NAME)
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.websocket_domain_name_from_env",
+                 return_value=TEST_WEBSOCKETS_DOMAIN_NAME)
+
+
+@pytest.fixture(name="mock_transaction_id")
+def mock_transaction_id_and_timeout_blocking(mocker):
+    mocker.patch("gsy_framework.client_connections.utils.uuid.uuid4",
+                 return_value=TEST_TRANSACTION_ID)
+
+
+@pytest.fixture(name="client")
+def fixture_rest_asset_client(mock_environment_use_functions):  # pylint: disable=unused-argument
+    return RestAssetClient(asset_uuid=TEST_ASSET_UUID)
+
+
+@pytest.mark.usefixtures("mock_environment_use_functions")
+class TestRestAssetClient:
+    """Test methods for RestAssetClient class."""
+
+    @staticmethod
+    @pytest.mark.parametrize("set_value, expected_ret_val",
+                             [(TEST_SIMULATION_ID, TEST_SIMULATION_ID),
+                              (None, TEST_SIMULATION_ID)])
+    def test_constructor_simulation_id_setup(set_value, expected_ret_val):
+        client = RestAssetClient(asset_uuid=TEST_ASSET_UUID,
+                                 simulation_id=set_value)
+        assert client.simulation_id == expected_ret_val
+
+    @staticmethod
+    @pytest.mark.parametrize("set_value, expected_ret_val",
+                             [(TEST_DOMAIN_NAME, TEST_DOMAIN_NAME),
+                              (None, TEST_DOMAIN_NAME)])
+    def test_constructor_domain_name_setup(set_value, expected_ret_val):
+        client = RestAssetClient(asset_uuid=TEST_ASSET_UUID,
+                                 domain_name=set_value)
+        assert client.domain_name == expected_ret_val
+
+    @staticmethod
+    @pytest.mark.parametrize("set_value, expected_ret_val",
+                             [(TEST_WEBSOCKETS_DOMAIN_NAME, TEST_WEBSOCKETS_DOMAIN_NAME),
+                              (None, TEST_WEBSOCKETS_DOMAIN_NAME)])
+    def test_constructor_websockets_domain_name_setup(set_value, expected_ret_val):
```

Comment:
I feel like the first 3 tests can be grouped in one single test, they're all testing whether the client will read the environment variables in case these configurations are not explicitly set (for instance in the first test you could've also tested the websocket_domain_name and it would've been set as well)
By the way, It would be better if you can pass values different than the ones in the environment variables, this will make tests relevant (making sure that the values you passed are prioritized over the environment ones)